### PR TITLE
Source artifacts

### PR DESCRIPTION
**Pull request by mnonnenmacher**
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/179)

### CODE REVIEW

**Comment by sschuberth** [comment]
Reviewed 8 of 8 files at r1.
Review status: 3 of 9 files reviewed at latest revision, 11 unresolved discussions.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 2 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-iNM_wbYZ6PR-zo6x:-L2-iNMaCYccVEJ7NOeV:b-8mxykz) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L2)):*
> ```Kotlin
> /*
>  * Copyright (c) 2017 HERE Europe B.V.
> ```

-2018

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 29 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-iviFZ8NcLucL_9Yn:-L2-iviGnA5wum1Stvv1:b-n9ncn9) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L29)):*
> ```Kotlin
> import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
> 
> private val TAR_GZ_EXTENSIONS = listOf(".tar.gz", ".tgz")
> ```

And ".tar.bz2", ".tbz2", if supported by Commons Compress.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 30 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-iWbFzLLNKZDeo9NA:-L2-iWbFzLLNKZDeo9NB:buubtln) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L30)):*
> ```Kotlin
> 
> private val TAR_GZ_EXTENSIONS = listOf(".tar.gz", ".tgz")
> private val ZIP_EXTENSIONS = listOf("jar", "war", "zip")
> ```

Maybe add ".aar", too? For Python, we'd also need ".egg" and ".whl" here.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 33 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-jRtNtELoetvhVckx:-L2-jRtNtELoetvhVcky:b-x9qktv) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L33)):*
> ```Kotlin
> 
> fun File.unpack(targetDirectory: File) {
>     when {
> ```

Maybe instead use `this.extension` in the `when` clause? However, `this.extension` doe snot include the leading dot, so more changes when defining the extensions above would be required.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 48 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-k1v_5kFNKNtNKQF-:-L2-k1v_5kFNKNtNKQF0:bjg2cfp) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L48)):*
> ```Kotlin
>     val gzipInputStream = GzipCompressorInputStream(inputStream())
>     val tarInputStream = TarArchiveInputStream(gzipInputStream)
>     tarInputStream.use {
> ```

Maybe remove / inline the `tarInputStream` variable altogether?

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 52 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-kLoMigyzl-07nN1D:-L2-kLoMigyzl-07nN1E:b8g4krc) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L52)):*
> ```Kotlin
>             val entry = it.nextTarEntry ?: break
> 
>             if (entry.isDirectory || entry.isSymbolicLink) {
> ```

Should we be even more strict and continue on everything that is not a regular file (i.e. also skip hardlinks, if they could occur)?

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 76 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-kyoLX4Ziy8ePT1uq:-L2-kyoLX4Ziy8ePT1ur:b-iub4av) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L76)):*
> ```Kotlin
> fun File.unpackZip(targetDirectory: File) {
>     val zipInputStream = ZipArchiveInputStream(inputStream())
>     zipInputStream.use {
> ```

Maybe remove / inline the `zipInputStream` variable altogether?

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 80 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-kcr_VRNW1Or9l55g:-L2-kcr_VRNW1Or9l55h:b-32ayhg) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L80)):*
> ```Kotlin
>             val entry = it.nextZipEntry ?: break
> 
>             if (entry.isDirectory || entry.isUnixSymlink) {
> ```

Similar to above, limit to regular files only?

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 102 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-lNGyw49Vu2fcUumH:-L2-lNGyw49Vu2fcUumI:b-w20try) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L102)):*
> ```Kotlin
>  * @return true if the parent directory could be created or if it already exists and is a directory.
>  */
> fun File.createParentDir(): Boolean {
> ```

Is it really worth to have this, or should we just call our existing `File.safeMkdirs()` on the parent?

---

*[downloader/src/test/assets/tmp/a, line 1 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-lga8_bZH573OGYAF:-L2-lga8_bZH573OGYAG:bh2cym4) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/test/assets/tmp/a#L1)):*
> ```
> a
> ```

I guess this file should not be committed.

---

*[downloader/src/test/assets/tmp/dir/b, line 1 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-lnBnO1DqoSE3SGmt:-L2-lnBnO1DqoSE3SGmu:bh2cym4) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/test/assets/tmp/dir/b#L1)):*
> ```
> b
> ```

I guess this file should not be committed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Comment by sschuberth** [comment]
Reviewed 1 of 1 files at r2.
Review status: 4 of 9 files reviewed at latest revision, 11 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Comment by sschuberth** [comment]
Reviewed 7 of 7 files at r3.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 34 at r3](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-mTG9gZS_ZP9FugbI:-L2-mTGAOPoWqs4oJ8tR:br1oh6w) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/6ebda72c5d48f8705e30caf98363346e3120ce8c/downloader/src/main/kotlin/ArchiveUtils.kt#L34)):*
> ```Kotlin
> fun File.unpack(targetDirectory: File) {
>     when {
>         TAR_GZ_EXTENSIONS.any { this.name.toLowerCase().endsWith(it) } -> unpackTarGz(targetDirectory)
> ```

These should be squashed into the first commit.

---

*[downloader/src/main/kotlin/Main.kt, line 257 at r3](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-mun5O1SxDYWvGlTn:-L2-mun6VaES65tD8UqL:b2yvoyy) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/6ebda72c5d48f8705e30caf98363346e3120ce8c/downloader/src/main/kotlin/Main.kt#L257)):*
> ```Kotlin
>         }
> 
>         val client = OkHttpClient()
> ```

Should use `OkHttpClientHelper`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Comment by mnonnenmacher** [comment]
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 2 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-iNM_wbYZ6PR-zo6x:-L2-y4uC6l49XT6z8H3m:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L2)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

-2018
</blockquote></details>

Done.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 29 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-iviFZ8NcLucL_9Yn:-L2-y64Q_znKe6px4daj:b8xib55) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L29)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

And ".tar.bz2", ".tbz2", if supported by Commons Compress.
</blockquote></details>

It's supported, but I haven't seen this used so far. I wanted to start with the formats I know are required and than add more on demand.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 30 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-iWbFzLLNKZDeo9NA:-L2-z4L1FGrIIAge2Pty:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L30)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Maybe add ".aar", too? For Python, we'd also need ".egg" and ".whl" here.
</blockquote></details>

Done.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 33 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-jRtNtELoetvhVckx:-L2-z82bamqzMU-Rn0We:b18cnx5) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L33)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Maybe instead use `this.extension` in the `when` clause? However, `this.extension` doe snot include the leading dot, so more changes when defining the extensions above would be required.
</blockquote></details>

File.extension uses substringAfterLast("."), so for example for tar.gz files it returns only "gz", not "tar.gz" which is what I need.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 48 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-k1v_5kFNKNtNKQF-:-L2-zQiq4KMTHRdv3iMD:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L48)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Maybe remove / inline the `tarInputStream` variable altogether?
</blockquote></details>

Done.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 52 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-kLoMigyzl-07nN1D:-L2-zfoCw9tc7pdoK2wR:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L52)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Should we be even more strict and continue on everything that is not a regular file (i.e. also skip hardlinks, if they could occur)?
</blockquote></details>

Done.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 76 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-kyoLX4Ziy8ePT1uq:-L2-zRpQW3SlplVAk3za:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L76)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Maybe remove / inline the `zipInputStream` variable altogether?
</blockquote></details>

Done.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 80 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-kcr_VRNW1Or9l55g:-L2-zuz9UNa-6D5oclDV:b-rhpef5) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L80)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Similar to above, limit to regular files only?
</blockquote></details>

For ZipArchiveEntry these are already all properties I can check.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 102 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-lNGyw49Vu2fcUumH:-L20-HmV6ixpIEAwEysc:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L102)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Is it really worth to have this, or should we just call our existing `File.safeMkdirs()` on the parent?
</blockquote></details>

Done.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 34 at r3](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-mTG9gZS_ZP9FugbI:-L2-zKX7hbRTUYfB4a5a:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/6ebda72c5d48f8705e30caf98363346e3120ce8c/downloader/src/main/kotlin/ArchiveUtils.kt#L34)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

These should be squashed into the first commit.
</blockquote></details>

Done.

---

*[downloader/src/main/kotlin/Main.kt, line 257 at r3](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-mun5O1SxDYWvGlTn:-L200l65MYnxD9TVySmt:b-7v1exh) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/6ebda72c5d48f8705e30caf98363346e3120ce8c/downloader/src/main/kotlin/Main.kt#L257)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Should use `OkHttpClientHelper`.
</blockquote></details>

I didn't want to use caching for the binary artifacts, do you think we should?

---

*[downloader/src/test/assets/tmp/a, line 1 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-lga8_bZH573OGYAF:-L20-krIHtCW9uz7bbC6:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/test/assets/tmp/a#L1)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

I guess this file should not be committed.
</blockquote></details>

Done.

---

*[downloader/src/test/assets/tmp/dir/b, line 1 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-lnBnO1DqoSE3SGmt:-L20-lFqU5tvhRo4m8jn:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/test/assets/tmp/dir/b#L1)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

I guess this file should not be committed.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Comment by sschuberth** [comment]
Review status: 3 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[downloader/src/main/kotlin/ArchiveUtils.kt, line 33 at r1](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-jRtNtELoetvhVckx:-L20BU_wBK5-rNEjhxvh:bdcv6l8) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/5dd58d18bafb1674ba8393b42b3db04845f5d9e4/downloader/src/main/kotlin/ArchiveUtils.kt#L33)):*
<details><summary><i>Previously, mnonnenmacher (Martin Nonnenmacher) wrote…</i></summary><blockquote>

File.extension uses substringAfterLast("."), so for example for tar.gz files it returns only "gz", not "tar.gz" which is what I need.
</blockquote></details>

Good point.

---

*[downloader/src/main/kotlin/Main.kt, line 257 at r3](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-mun5O1SxDYWvGlTn:-L20Bt-xUAZ6W5zoE5Ms:b2z8uaw) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/6ebda72c5d48f8705e30caf98363346e3120ce8c/downloader/src/main/kotlin/Main.kt#L257)):*
<details><summary><i>Previously, mnonnenmacher (Martin Nonnenmacher) wrote…</i></summary><blockquote>

I didn't want to use caching for the binary artifacts, do you think we should?
</blockquote></details>

Well, why not? Are you afraid of the cache growing too large?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Comment by mnonnenmacher** [comment]
Review status: 3 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[downloader/src/main/kotlin/Main.kt, line 257 at r3](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-mun5O1SxDYWvGlTn:-L20CRQ5TPBmvCnY-EeV:b-k8k3z) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/6ebda72c5d48f8705e30caf98363346e3120ce8c/downloader/src/main/kotlin/Main.kt#L257)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Well, why not? Are you afraid of the cache growing too large?
</blockquote></details>

Partly this, and that we usually archive the scan results and download any source code twice. But for debugging and when the scan fails caching could still help, so I'll change it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Comment by mnonnenmacher** [comment]
Review status: 3 of 11 files reviewed at latest revision, 2 unresolved discussions.

---

*[downloader/src/main/kotlin/Main.kt, line 257 at r3](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179#-L2-mun5O1SxDYWvGlTn:-L20DoFZx_yOuCsySOe5:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/6ebda72c5d48f8705e30caf98363346e3120ce8c/downloader/src/main/kotlin/Main.kt#L257)):*
<details><summary><i>Previously, mnonnenmacher (Martin Nonnenmacher) wrote…</i></summary><blockquote>

Partly this, and that we usually archive the scan results and download any source code twice. But for debugging and when the scan fails caching could still help, so I'll change it.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

File: downloader/src/main/kotlin/ArchiveUtils.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import java.io.File
+import java.io.IOException
+
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
+
+private val TAR_GZ_EXTENSIONS = listOf(".tar.gz", ".tgz")
+private val ZIP_EXTENSIONS = listOf(".aar", ".egg", ".jar", ".war", ".zip")
```
Comment (sschuberth) [inline downloader/src/main/kotlin/ArchiveUtils.kt]: ".whl" is still missing.

File: downloader/src/main/kotlin/Main.kt [hunk 2]
```diff
@@ -237,9 +244,54 @@ object Main {
...
+            throw DownloadException(e)
+        }
+
+        return outputDirectory
+    }
+
+    private fun verifyChecksum(file: File, hash: String, hashAlgorithm: String) {
+        val digest = when (hashAlgorithm.toLowerCase()) {
+            "md5" -> file.inputStream().use { DigestUtils.md5Hex(it) }
+            "sha-1" -> file.inputStream().use { DigestUtils.sha1Hex(it) }
```
Comment (sschuberth) [inline downloader/src/main/kotlin/Main.kt]: Note: We probably need to also support "sha1" (without the dash) here in the future. Maybe add that already now, like `"sha1", "sha-1" ->`.

File: downloader/src/main/kotlin/ArchiveUtils.kt (hunk 1 above)
Comment (mnonnenmacher) [reply downloader/src/main/kotlin/ArchiveUtils.kt]: Done.

File: downloader/src/main/kotlin/Main.kt (hunk 2 above)
Comment (mnonnenmacher) [reply downloader/src/main/kotlin/Main.kt]: Didn't know that syntax, cool.

**Comment by sschuberth** [comment]
Reviewed 8 of 8 files at r7.
Review status: 4 of 11 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Comment by sschuberth** [comment]
Reviewed 1 of 1 files at r8.
Review status: 5 of 11 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Comment by sschuberth** [comment]
Reviewed 6 of 6 files at r9.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/179)*

**Pull request merged**